solutions/silver/usaco-645.mdx

```diff
@@ -112,6 +112,97 @@ int main() {
 }
 ```
 </CPPSection>
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+class Pair implements Comparable<Pair> {
+	public long first, second;
+	public Pair(long x, long y) {
+		first = x;
+		second = y;
+	}
+	public int compareTo(Pair o) {
+		if (o.first != first) return (int)(first - o.first);
+		return (int)(second - o.second);
+	}
+}
+
+public class Main {
+
+	// returns updated min/maxs when considering a new coordinate
+	public static Pair upd(Pair a, long b) {
+		return new Pair(Math.min(a.first, b), Math.max(a.second, b));
+	}
+
+	public static void main(String[] args) throws Exception {
+		BufferedReader br = new BufferedReader(new FileReader("split.in"));
+		StringTokenizer st;
```

Comment:
declare this in the for loop

Reply:
Why? Isn't it better to declare one and initialize several times?

Reply:
is it? can i have a so link if that's the case